R/messages.R

```diff
@@ -0,0 +1,31 @@
+messages <- list(
+  errorWrongType = function(objectName, type, expectedType, optionalMessage = NULL) {
+    # Name of the calling function
+    callingFunctions <- sys.calls()
+    callingFunction <- sys.call(-length(callingFunctions) + 1)[[1]]
+
+    expectedTypeMsg <- paste0(expectedType, collapse = ", or ")
+
+    paste0(
+      callingFunction, ": argument '", objectName,
+      "' is of type '", type, "', but expected '", expectedTypeMsg, "'!", optionalMessage
+    )
+  },
+
+  errorDifferentLength = function(objectNames, optionalMessage = NULL) {
+    # Name of the calling function
+    callingFunctions <- sys.calls()
+    callingFunction <- sys.call(-length(callingFunctions) + 1)[[1]]
+
+    paste0(
+      callingFunction, ": Arguments '", objectNames,
+      "' must have the same length, but they don't!", optionalMessage
+    )
+  },
+
+  errorEnumNotAllNames = "The enumValues has some but not all names assigned. They must be all assigned or none assigned",

```

Comment:
Example of usage: Advantage. We do not repeat ourselves and all messages are localized so they are easy to change